libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/plugin/BuildEventsListenerRegistryHolder.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.gradle.plugin
+
+import org.gradle.api.Project
+import org.gradle.build.event.BuildEventsListenerRegistry
+import org.jetbrains.kotlin.gradle.utils.isConfigurationCacheAvailable
+import javax.inject.Inject
+
+open class BuildEventsListenerRegistryHolder @Inject constructor(val listenerRegistry: BuildEventsListenerRegistry?) {
+    companion object {
+        fun getInstance(project: Project) = run {
+            if (isConfigurationCacheAvailable(project.gradle)) {
```

Comment:
:x: the check should be `gradleVersion >= 6.1` instead where `BuildEventsListenerRegistry` is available

Reply:
New listener is not ready for configuration cache